Assets/Scripts/Gameplay/GameplayObjects/ServerTossAttack.cs

```diff
@@ -0,0 +1,103 @@
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Server
+{
+    public class ServerTossAttack : NetworkBehaviour
+    {
+        [SerializeField]
+        int m_DamagePoints;
+
+        [SerializeField]
+        float m_HitRadius = 5f;
+
+        [SerializeField]
+        float m_KnockbackSpeed;
+
+        [SerializeField]
+        float m_KnockbackDuration;
+
+        [SerializeField]
+        LayerMask m_LayerMask;
+
+        bool m_Started;
+
+        const int k_MaxCollisions = 16;
+        Collider[] m_CollisionCache = new Collider[k_MaxCollisions];
+
+        [SerializeField]
+        float detonateAfterSeconds = 5f;
+
+        float m_DetonateAfterSeconds;
+
+        [SerializeField]
+        float destroyAfterSeconds = 5.5f;
+
+        float m_DestroyAfterSeconds;
+
+        bool m_Detonated;
+
+        public override void OnNetworkSpawn()
+        {
+            if (!IsServer)
+            {
+                enabled = false;
+                return;
+            }
+
+            m_Started = true;
+            m_Detonated = false;
+
+            m_DetonateAfterSeconds = Time.fixedTime + detonateAfterSeconds;
+            m_DestroyAfterSeconds = Time.fixedTime + destroyAfterSeconds;
```

Comment:
why is destroy different from detonate? why don't both happen at the same time? If it's just for the explosion VFX, couldn't it just spawn that VFX client side and destroy server side?

Reply:
Right. We can have the server send the detonate message, and clients can spawn a detonation VFX at that instant. Will adjust.